cmd/chisel/cmd_cut_test.go

```diff
@@ -0,0 +1,426 @@
+package main_test
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/klauspost/compress/zstd"
+	. "gopkg.in/check.v1"
+
+	chisel "github.com/canonical/chisel/cmd/chisel"
+	"github.com/canonical/chisel/internal/archive"
+	"github.com/canonical/chisel/internal/testutil"
+)
+
+var (
+	testKey = testutil.PGPKeys["key1"]
+)
+
+type cutTest struct {
+	summary    string
+	release    map[string]string
+	slices     []string
+	pkgs       map[string][]byte
+	filesystem map[string]string
+	db         string
+	dbPaths    []string
+	err        string
+}
+
+var cutTests = []cutTest{{
+	summary: "Basic cut",
+	release: map[string]string{
+		"slices/mydir/test-package.yaml": `
+			package: test-package
+			slices:
+				myslice:
+					contents:
+						/dir/file:
+						/dir/file-copy:  {copy: /dir/file}
+						/other-dir/file: {symlink: ../dir/file}
+						/dir/text-file:  {text: data1}
+						/dir/foo/bar/:   {make: true, mode: 01777}
+				manifest:
+					contents:
+						/db/**: {generate: manifest}
+		`,
+	},
+	slices: []string{"test-package_myslice", "test-package_manifest"},
+	filesystem: map[string]string{
+		"/db/":            "dir 0755",
+		"/db/chisel.db":   "file 0644 1769a2ce",
+		"/dir/":           "dir 0755",
+		"/dir/file":       "file 0644 cc55e2ec",
+		"/dir/file-copy":  "file 0644 cc55e2ec",
+		"/dir/foo/":       "dir 0755",
+		"/dir/foo/bar/":   "dir 01777",
+		"/dir/text-file":  "file 0644 5b41362b",
+		"/other-dir/":     "dir 0755",
+		"/other-dir/file": "symlink ../dir/file",
+	},
+	dbPaths: []string{"/db/chisel.db"},
+	db: `
+{"jsonwall":"1.0","schema":"1.0","count":16}
+{"kind":"content","slice":"test-package_manifest","path":"/db/chisel.db"}
+{"kind":"content","slice":"test-package_myslice","path":"/dir/file"}
+{"kind":"content","slice":"test-package_myslice","path":"/dir/file-copy"}
+{"kind":"content","slice":"test-package_myslice","path":"/dir/foo/bar/"}
+{"kind":"content","slice":"test-package_myslice","path":"/dir/text-file"}
+{"kind":"content","slice":"test-package_myslice","path":"/other-dir/file"}
+{"kind":"package","name":"test-package","version":"test-package_version","sha256":"test-package_hash","arch":"test-package_arch"}
+{"kind":"path","path":"/db/chisel.db","mode":"0644","slices":["test-package_manifest"]}
+{"kind":"path","path":"/dir/file","mode":"0644","slices":["test-package_myslice"],"sha256":"cc55e2ecf36e40171ded57167c38e1025c99dc8f8bcdd6422368385a977ae1fe","size":14}
+{"kind":"path","path":"/dir/file-copy","mode":"0644","slices":["test-package_myslice"],"sha256":"cc55e2ecf36e40171ded57167c38e1025c99dc8f8bcdd6422368385a977ae1fe","size":14}
+{"kind":"path","path":"/dir/foo/bar/","mode":"0777","slices":["test-package_myslice"]}
```

Comment:
Hmm... it can be, yes. In that case, we will need to check if the path we are adding is a directory path.
Let me know if you wanna go ahead with this, or ask Gustavo for further clars?

Reply:
If you can use `mode: 01777` is the yaml, which you can, then it should be also in the db, in my opinion. Let's add it now even if requires the custom bit mask.